test/src/test/java/hudson/search/SearchTest.java

```diff
@@ -225,6 +258,91 @@ else if(displayName.equals(name)) {
         assertTrue(foundDispayName);
     }
 
+    @Issue("JENKINS-24433")
+    @Test
+    public void testProjectNameBehindAFolderDisplayName() throws Exception {
+        final String projectName1 = "job-1";
+        final String displayName1 = "job-1";
+
+        final String projectName2 = "job-2";
+        final String displayName2 = "job-2";
+
+        FreeStyleProject project1 = j.createFreeStyleProject(projectName1);
+        project1.setDisplayName(displayName1);
+
+        FreeStyleProject project2 = j.createFolder("my-folder-1").createProject(FreeStyleProject.class, projectName2);
+        project2.setDisplayName(displayName2);
+
+        WebClient wc = j.createWebClient();
+        Page result = wc.goTo("job/my-folder-1/search/suggest?query=" + projectName1, "application/json");
+        assertNotNull(result);
+        j.assertGoodStatus(result);
+
+        String content = result.getWebResponse().getContentAsString();
+        JSONObject jsonContent = (JSONObject)JSONSerializer.toJSON(content);
+        assertNotNull(jsonContent);
+        JSONArray jsonArray = jsonContent.getJSONArray("suggestions");
+        assertNotNull(jsonArray);
+
+        assertEquals(1, jsonArray.size());
+
+        boolean foundProjectName = false;
+        boolean foundDisplayName = false;
+        for(Object suggestion : jsonArray) {
+            JSONObject jsonSuggestion = (JSONObject)suggestion;
+
+            String name = (String)jsonSuggestion.get("name");
+            if(displayName1.equals(name)) {
+                foundDisplayName = true;
+            }
+        }
+
+        assertTrue(foundDisplayName);
+    }
+
+    @Issue("JENKINS-24433")
+    @Test
+    public void testProjectNameInAFolderDisplayName() throws Exception {
+        final String projectName1 = "job-1";
+        final String displayName1 = "job-1";
+
+        final String projectName2 = "job-2";
+        final String displayName2 = "my-folder-1 job-2";
+
+        FreeStyleProject project1 = j.createFreeStyleProject(projectName1);
+        project1.setDisplayName(displayName1);
+
+        FreeStyleProject project2 = j.createFolder("my-folder-1").createProject(FreeStyleProject.class, projectName2);
+        project2.setDisplayName(displayName2);
+
+
+        WebClient wc = j.createWebClient();
+        Page result = wc.goTo("job/my-folder-1/search/suggest?query=" + projectName2, "application/json");
```

Comment:
Again could use `[MockFolder.]getUrl()` to make it a little clearer where this URL is coming from.